File: ps/web/HTML/flash.php
<?php

$fwidth = isset($_POST['fwidth']) ? $_POST['fwidth'] : '';
$fheight = isset($_POST['fheight']) ? $_POST['fheight'] : '';
$ftitle = isset($_POST['ftitle']) ? $_POST['ftitle'] : '';
$ffile = isset($_POST['ffile']) ? $_POST['ffile'] : '';
$product = isset($_POST['product']) ? $_POST['product'] : '';
$sourcepage = isset($_POST['sourcepage']) ? $_POST['sourcepage'] : '';
$recepient = "flash@powersoftware.com";

$cr = "\r\n";

$to = $recepient;

$headers = "From: $recepient $cr";
$headers .= "MIME-Version: 1.0$cr";
$headers .= "Content-Type: text/html; charset=ISO-8859-1$cr";

$subject = "Flash Viewing of $product";

$message = "<html><head>$cr".
"<style type=\"text/css\">$cr".
"td,p,li {font-family:Arial,Sans Serif;font-size:11pt;}$cr".
"b {color:#666666}$cr".
"b.server {color:#6666FF}$cr".
"p.cpyrght {font-size:9pt; color:#6666FF;text-align:center;}$cr".
"</style></head><body>$cr".
"<p>A user viewed the <b>$product</b> Flash Presentation on " . strftime("<b>%a %d %b</b> at <b>%H:%M</b>"). ".</p>$cr$cr".
"<ul><li>Page <b>$sourcepage</b>.</li>$cr$cr".
"<li>Server <b class=server>" . $_SERVER[SERVER_NAME] . "</b>.</li>$cr$cr".
"<li>IP address <b>" . $_SERVER[REMOTE_ADDR] . "</b>.</li>$cr$cr".
"<li>Browser <b>" . $_SERVER[HTTP_USER_AGENT] . "</b>.</li></ul>$cr$cr".
"<p class=cpyrght>Copyright &copy; " . strftime("%Y"). " PowerSoftware.com.  All Rights Reserved.</p>$cr$cr".
"</body></html>";

mail($to,$subject,$message,$headers);

/*
<OBJECT classid="clsid:D27CDB6E-AE6D-11cf-96B8-444553540000"
        codebase="http://download.macromedia.com/pub/shockwave/cabs/flash/swflash.cab#version=5,0,0,0"
        WIDTH="<? echo $fwidth; ?>" HEIGHT="<? echo $fheight; ?>">
  <PARAM NAME=movie VALUE="<? echo $fheight; ?>">
  <PARAM NAME=loop VALUE=false>
  <PARAM NAME=quality VALUE=high>
  <PARAM NAME=bgcolor VALUE=#FFFFFF>
  <EMBED src="<? echo $ffile; ?>" loop=false quality=high bgcolor=#FFFFFF
    WIDTH="<? echo $fwidth; ?>" HEIGHT="<? echo $fheight; ?>"
    TYPE="application/x-shockwave-flash"
    PLUGINSPAGE="http://www.macromedia.com/shockwave/download/index.cgi?P1_Prod_Version=ShockwaveFlash"></EMBED>
</OBJECT>
*/

?>
<html>
<head><title><? echo $ftitle; ?></title></head>
<body bgcolor="#FFFFFF" marginwidth=0 marginheight=0 topmargin=0 leftmargin=0 rightmargin=0>

<object classid="clsid:D27CDB6E-AE6D-11cf-96B8-444553540000"
        codebase="http://download.macromedia.com/pub/shockwave/cabs/flash/swflash.cab#version=7,0,19,0"
        width="<? echo $fwidth; ?>" height="<? echo $fheight; ?>" title="<? echo $product; ?>">
      <param name="movie" value="<? echo $ffile; ?>" />
      <param name="quality" value="high" />
      <embed src="<? echo $ffile; ?>" quality="high"
        pluginspage="http://www.macromedia.com/go/getflashplayer"
        type="application/x-shockwave-flash" width="<? echo $fwidth; ?>" height="<? echo $fheight; ?>"></embed>
    </object>

</body>
</html>

File: ps/dev/web/cgi-bin/flash.pl
#!/usr/bin/perl

print "Content-type: text/html\n\n" ;

# This should match the mail program on your system.
$mailprog = '/usr/sbin/sendmail' ;

# This should be set to the username or alias that runs your
# WWW server.
$recipient = 'flash@powersoftware.com' ;

$fwidth =  "fwidth" ;
$fheight = "fheight" ;
$ftitle =  "ftitle" ;
$ffile =   "ffile" ;

# Get the input
read(STDIN, $buffer, $ENV{'CONTENT_LENGTH'}) ;

print "<html>" ;

# Split the name-value pairs
@pairs = split(/&/, $buffer) ;

foreach $pair (@pairs)
{
    ($name, $value) = split(/=/, $pair) ;

    # Un-Webify plus signs and %-encoding
    $value =~ tr/+/ /;
    $value =~ s/%([a-fA-F0-9][a-fA-F0-9])/pack("C", hex($1))/eg ;

    # Uncomment for debugging purposes
    #print "<p>Setting $name to $value</p>" ;

    $FORM{$name} = $value ;
}

# Now send mail to $recipient

$product = $FORM{'product'} ;

open (MAIL, "|$mailprog $recipient") || die "Can't open $mailprog!\n" ;
print MAIL "Subject: Visitor viewing Flash - ".$product."\n\n" ;
print MAIL $product."\n\n" ;
print MAIL "User's browser:    $ENV{'HTTP_USER_AGENT'}\n" ;
print MAIL "Remote host:       $ENV{'REMOTE_HOST'}\n" ;
print MAIL "Remote IP address: $ENV{'REMOTE_ADDR'}\n\n" ;
close (MAIL) ;

print "<head><title>".$FORM{$ftitle}."</title></head><body bgcolor=\"#FFFFFF\" marginwidth=0 marginheight=0 topmargin=0 leftmargin=0 rightmargin=0>" ;

print "<OBJECT classid=\"clsid:D27CDB6E-AE6D-11cf-96B8-444553540000\"" ;
print " codebase=\"http://download.macromedia.com/pub/shockwave/cabs/flash/swflash.cab#version=5,0,0,0\"" ;
print " WIDTH=".$FORM{$fwidth}. " HEIGHT=".$FORM{$fheight}.">" ;
print "<PARAM NAME=movie VALUE=\"".$FORM{$ffile}."\">" ;
print "<PARAM NAME=loop VALUE=false><PARAM NAME=quality VALUE=high>" ;
print "<PARAM NAME=bgcolor VALUE=#FFFFFF><EMBED src=\"".$FORM{$ffile}."\" loop=false quality=high bgcolor=#FFFFFF WIDTH=".$FORM{$fwidth}." HEIGHT=".$FORM{$fheight}." TYPE=\"application/x-shockwave-flash\" PLUGINSPAGE=\"http://www.macromedia.com/shockwave/download/index.cgi?P1_Prod_Version=ShockwaveFlash\"></EMBED>" ;
print "</OBJECT>" ;

print "</body></html>" ;
exit ;